src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTask.java

```diff
@@ -106,11 +106,16 @@ public boolean ignoreRecord(final SinkRecord record) {
     }
 
     private void tryWriteRecord(final SinkRecord record) {
-        final var index = convertTopicToIndexName(record.topic());
-        ensureIndexExists(index);
-        checkMappingFor(index, record);
+        final var indexOrDataStreamName =
+                config.dataStreamEnabled()
```

Comment:
This logic could be moved off to constructor (since `config.dataStreamEnabled()` is global and does not depend on the record, something like that:

```
indexNameProvider = config.dataStreamEnabled() ? 
    (name) -> /* data stream name generator */ 
        : this::convertTopicToIndexName;
```

Reply:
the name depends on the record which contains info about topic

Reply:
That is why the function has `(name)` argument, but the point is - the sink either uses data streams or index for all records - this decision could be taken upfront, no need to revisit it on every record.

Reply:
i moved logic to `convertTopicToIndexName` and renamed function to `convertTopicToIndexOrDataStreamName`

Reply:
done moved logic to the confg